Algorithm.Framework/Alphas/MacdAlphaModel.cs

```diff
@@ -30,22 +30,31 @@ namespace QuantConnect.Algorithm.Framework.Alphas
     /// </summary>
     public class MacdAlphaModel : IAlphaModel
     {
-        private readonly TimeSpan _insightPeriod;
-        private readonly TimeSpan _consolidatorPeriod;
-        private readonly decimal _bounceThresholdPercent;
+        private readonly int _fastPeriod;
+        private readonly int _slowPeriod;
+        private readonly int _signalPeriod;
+        private readonly MovingAverageType _movingAverageType;
+        private const decimal BounceThresholdPercent = 0.01m;
         private readonly Dictionary<Symbol, SymbolData> _symbolData;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="MacdAlphaModel"/> class
         /// </summary>
-        /// <param name="consolidatorPeriod">The period of the MACD's input</param>
-        /// <param name="insightPeriod">The period assigned to generate insight</param>
-        /// <param name="bounceThresholdPercent">The percent change required in the MACD signal line to warrant an up/down insight</param>
-        public MacdAlphaModel(TimeSpan consolidatorPeriod, TimeSpan insightPeriod, decimal bounceThresholdPercent)
+        /// <param name="fastPeriod">The MACD fast period</param>
+        /// <param name="slowPeriod">The MACD slow period</param>
+        /// <param name="signalPeriod">The smoothing period for the MACD signal</param>
+        /// <param name="movingAverageType">The type of moving average to use in the MACD</param>
+        public MacdAlphaModel(
+            int fastPeriod = 12,
```

Comment:
Updated constructor args to explicitly match the indicator arguments. Made them all inside 1 class file to make the alpha models easier to digest in a single take.